index.js

```diff
@@ -19,24 +29,43 @@ const cacheReply = function(fn, cacheOptions = {}) {
     // return cache stats if requested:
     if (query[statsQueryParam]) {
       return {
+        statusCode: 200,
         headers: {
           'content-type': 'application/json; charset=utf8'
         },
-        body: JSON.stringify(cache.cache.getStats())
+        body: JSON.stringify(cache.getStats())
       };
     }
-    // skip cache altogether if requested:
-    if (query[skipQueryParam]) {
+    // skip cache altogether if requested or explicitly disabled:
+    if (query[skipQueryParam] || cacheOptions.enabled === false) {
+      if (cacheOptions.log) {
+        cacheOptions.log(['cache', 'skip'], { key: memoKey, query });
+      }
       return fn(req);
     }
     // return the cached value:
-    return cache.memo(
+    const result = await memo(
       memoKey,
-      () => fn(req),
+      () => {
+        // log misses:
+        if (cacheOptions.log && !query[dropQueryParam]) {
+          cacheOptions.log(['cache', 'miss'], { key: memoKey, query });
+        }
+        return fn(req);
+      },
       ttl,
       query[dropQueryParam] // force-update cache if true
     );
+    // log if update or hit:
+    if (cacheOptions.log) {
+      if (query[dropQueryParam]) {
```

Comment:
might be cleaner to just bring memo logic inside this function so we have access to log